### ISSUE TITLE
[Request] Allow exporting with a different password

### ISSUE THREAD

**Issue by SlugFiller** [qTox/qTox#3695]
From what I can tell, the "Export" feature in qTox is roughly equivalent to copying the profile and chat log files around. This seems redundant, but maybe that's just me.

What would be useful, however, is to be able to export the file under a different (potentially more secure) password. The rationale is that if you're backing up your profile on a less secure storage (DropBox, Google Drive, random FTP server, etc), you can compensate for the loss of storage security with increase in password security.

Before anyone says "Just always use the more secure password", consider that security always comes at the cost of convenience. On a secure local storage, and under daily use, a higher entropy password may be less justified than in a remote backup that you would only use in case of emergency. Different scenarios prompt for different levels of security. And it would be nice to be able to fine tune it without having to change the password and re-encrypt back and forth each time.

**Comment by zetok** [comment]
How should UI for it look like?

**Comment by SlugFiller** [comment]
Similar to what you have today. You hit "Export", and either before or after the "Save As" dialog, it opens a dialog that asks you to type in a password for the exported file. You type a password, and it saves/exports with that password.

For extra credit, it could have a "confirm password" field as well.

**Comment by zetok** [comment]
>You hit "Export", and either before or after the "Save As" dialog, it opens a dialog that asks you to type in a password for the exported file.

Why someone should need to type in again password if they already have one?
Also, why is there a need for an additional pop-up dialog?

**Comment by SlugFiller** [comment]
The idea is to type in a DIFFERENT password (or be able to type in one even if they don't have one).

As I've explained in the opening post, people might want a different level of security for their local Tox installation (which has to be unlocked on every execution, but is only accessible to people with access to your harddrive), and a remote backup (which only needs to be unlocked if you're recovering from backup, but is easily accessible by your backup provider). These are two completely different scenarios, with different security concerns. So the idea that you'd want the exact same password for both is overly presumptuous.

And since there's no tool for "changing the password" of a backup, the only "solution" currently is to change the password on your account, export, then change the password back.

Besides, if copying the account as-is was the only use case, the export button as a whole would be superficial. You can just copy the files.

**Comment by zetok** [comment]
>The idea is to type in a DIFFERENT password (or be able to type in one even if they don't have one).
>
>As I've explained in the opening post, people might want a different level of security for their local Tox installation (which has to be unlocked on every execution, but is only accessible to people with access to your harddrive), and a remote backup (which only needs to be unlocked if you're recovering from backup, but is easily accessible by your backup provider). These are two completely different scenarios, with different security concerns. So the idea that you'd want the exact same password for both is overly presumptuous.

Yes, but why would you make life harder for people who are perfectly fine with using their current password, and most likely would be the majority of use-cases?

>Besides, if copying the account as-is was the only use case, the export button as a whole would be superficial. You can just copy the files.

Indeed you can, and the button is just a part of an easy-to-use UI.

**Comment by SlugFiller** [comment]
It's easy enough to add a "use existing password" button, if people who type that password every time they run Tox suddenly get lazy the once-in-a-century where they need to re-export their account.

And, as you've noted, even before that, the export, currently, is just "sugar" for an operation that is already trivial. While what I'm suggesting is something that is actually difficult to do, and cannot be practically done without support.

Unless there's an external tool I'm not aware of for "re-passwording" an export, this is something that needs to be in the client. It's not merely "sugar".

**Comment by zetok** [comment]
>It's easy enough to add a "use existing password" button, if people who type that password every time they run Tox suddenly get lazy the once-in-a-century where they need to re-export their account.

It's not about adding a button, but about making UI worse.

There's no need to retype the password in the first place. The only reason why retyping password would be needed is when someone would want to export with a different password, something that majority of users most likely wouldn't do.

And yes, people are lazy.

>And, as you've noted, even before that, the export, currently, is just "sugar" for an operation that is already trivial.

Trivial for some, not so trivial for others.

>While what I'm suggesting is something that is actually difficult to do, and cannot be practically done without support.

Right. There is a problem with current form of the proposal in that it makes UI worse.

>Unless there's an external tool I'm not aware of for "re-passwording" an export, this is something that needs to be in the client. It's not merely "sugar".

Tool like that would be easy to make.

----
Anyways; the idea for `export` button without making UI worse:
![spectacle wl2573](https://cloud.githubusercontent.com/assets/3148759/20861566/4948c3e6-b98b-11e6-9491-c633e24e71a6.png)
↑ the above button used in `spectacle` has a default action, and a drop-down list with stuff other than the default action.

`Export` should be the default button, and the drop-down list should offer `Export with a new password`.

**Comment by SlugFiller** [comment]
That seems reasonable. Anything is better than the current state, in which it is simply impossible.
